tools/pipelines/templates/include-generate-notice-steps.yml

```diff
@@ -10,12 +10,11 @@ parameters:
 steps:
 # special case server/routerlicious
 - ${{ if eq(parameters.buildDirectory, 'server/routerlicious') }}:
-  - task: Npm@1
-    displayName: npm ci
-    inputs:
-      command: 'custom'
-      customCommand: 'ci --ignore-scripts'
-      customRegistry: 'useNpmrc'
+  - bash: |
+      npm i -g pnpm
+      pnpm i --frozen-lockfile
+    displayName: 'Install deps'
```

Comment:
Why does the routerlicious pipeline need to install the root deps?

Reply:
I'd like to know that too haha. It seems to be just so it can use `node_modules/@fluidframework/build-tools/dist/genMonoRepoPackageJson/genMonoRepoPackageJson.js --server`. server/routerlicious has its own (outdated) dependency on build-tools so we could also tell it to run the script from there (I tried and the pipeline [didn't like something](https://dev.azure.com/fluidframework/internal/_build/results?buildId=124884&view=logs&j=3dc8fd7e-4368-5a92-293e-d53cefc8c4b3&t=996a989a-edd9-5084-d9de-d81350de1bfc), seemed to me like the script failed to generate the expected output, but maybe it needs to be called differently from that folder vs from the root).

Reply:
And related, I updated the build-tools dependency but then [tests fail](https://dev.azure.com/fluidframework/internal/_build/results?buildId=124891&view=logs&j=3dc8fd7e-4368-5a92-293e-d53cefc8c4b3&t=138573f1-1c81-5936-fd5c-74f32fbb8748) and for now I just wanted to get the pipeline working again.

Reply:
Also, I think if we do this special casing so component detection runs for server/routerlicious, we probably should do it for server/historian and server/gitrest too. @curtisman as the original author of the special-case for server/routerlicious, any insight into why only that one has it?